File: 백준/Gold/11729. 하노이 탑 이동 순서/하노이 탑 이동 순서.cc
#include <iostream>
#include <cmath>
using namespace std;

void hanoi(int n, int one, int two, int three)
{
	if (n == 1)
		cout << one << " " << three << "\n";
	else
	{
		hanoi(n - 1, one, three, two);
		cout << one << " " << three << "\n";
		hanoi(n - 1, two, one, three);
	}
}

int main()
{
	int n, cnt = 0;
	cin >> n;
	cout << (int)pow(2, n) - 1 << "\n";
	hanoi(n, 1, 2, 3);
	return 0;
}

File: 백준/Gold/11729. 하노이 탑 이동 순서/하노이 탑 이동 순서.py
#a,b,c는 각각의 기둥
def hanoi(n,a,b,c):
    if n==1:
        print(a,c)
    else:
        hanoi(n-1,a,c,b)
        print(a,c)
        hanoi(n-1,b,a,c)

N = int(input())
#원판의 총 이동 개수
print(2**N-1)
hanoi(N,1,2,3)